sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -501,9 +501,8 @@ object CatalogColumnStat extends Logging {
    */
   def toExternalString(v: Any, colName: String, dataType: DataType): String = {
     val externalValue = dataType match {
-      case DateType => DateTimeUtils.toJavaDate(v.asInstanceOf[Int])
-      case TimestampType => DateTimeUtils.toJavaTimestamp(v.asInstanceOf[Long])
-      case BooleanType | _: IntegralType | FloatType | DoubleType => v
+      case BooleanType | _: IntegralType | FloatType | DoubleType |
+           TimestampType | DateType => v
```

Comment:
If we do this, then the stats string will not be human-readable. Shall we still use a timestamp/date format, but put the timezone at the end?

Reply:
> If we do this, then the stats string will not be human-readable.

Is it necessary to be like human-readable? 

> Shall we still use a timestamp/date format, but put the timezone at the end?

If I add timezone at the end, I am not sure the stored text can be more readable from meaning perspective. For example, you have `2019-01-28 01:00:00` in your time zone `UTC+8` but when I see that I have to subtract 8 hours to get real timestamp. Taking into account that it goes over the end of the day. 

Unix timestamp can be easily converted to text, just google `unix time` - there are a lot of websites.

Reply:
> Is it necessary to be like human-readable?

AFAIK this is the design goal when we implement CBO, cc @rxin @wzhfy 

> If I add timezone at the end, I am not sure the stored text can be more readable from meaning perspective

How about we store the timestamp string in UTC? Then the read path can pick the right timezone based on the version number: if it's version 2, convert the timestamp string to long with UTC. If it's version 1, do the conversion with session local timezone, which might be wrong but it's the best effort.